### ISSUE TITLE
Test if the needed functions are enabled

### ISSUE THREAD

**Issue by adrielairaldo** [mikehaertl/php-shellcommand#50]
Hi,

I'm trying your php-pdftk. I've been receiving some errors, and I found out the problem was related to this library. It took me some time to discover the problem was the lack of enabled functions. Perhaps having an error message notifying that the needed functions are not enabled may save some time. I had to scan the code to see which functions were necessary (exec, proc_open, proc_close, proc_get_status, proc_terminate, escapeshellarg, escapeshellcmd)

In general, I just use something like the next code to discover if a function is present:

```
    if(!function_exists('exec')) {
        echo "exec is not enabled"; // Do something relevant here
    }
```
I'm just an occasional PHP developer, sorry if I'm suggesting something silly.

Thank a lot for all your libraries, you are amazing!

**Comment by mikehaertl** [comment]
While I understand your trouble I think that checking for these functions would not help much. If you get error messages they should already point you to the right direction. All our own checks could do is to report a similar error so there's no benefit in having these checks. Of course there could be some sort of requirement checker - but for a simple library like this I find this would be overkill.

Also AFAIK `exec()` is part of the PHP core and enabled by default so your situation should not be very common. In fact it's the first report I see on this issue.

So closing this as I don't think there's any reasonable solution here.

**Comment by adrielairaldo** [comment]
Thanks for your time. Just adding some extra info a personal perspective.

I just got a "Could not run pdftk", nothing else. I had to inspect the code, and then I saw those calls, and I went to the php.ini to change the configuration. 

As far as I have seen, servers are always deployed with the criteria of "granting the minimum necessary permissions", and all those functions are blocked for obvious reasons (the full access to your console). On the other hand, hosting-only systems (not dedicated servers) clearly do not provide this type of functionality.

Platforms as Nextcloud, for example, tell you about the lack of available needed commands on your system to fully work.

Again, thanks for your work, it's amazing and works flawlessly!
